prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -175,6 +182,7 @@ class FHIRRouter(
         actionHistory: ActionHistory
     ) {
         logger.trace("Processing HL7 data for FHIR conversion.")
+        this.actionLogger = actionLogger
```

Comment:
I feel like this should be set at the constructor if you are setting a class property. Can that be done?

Reply:
I don't see an easy way to do this without refactoring on FHIREngine and the other classes that inherit from it